src/configuration/prima.rs

```diff
@@ -10,17 +10,25 @@ pub struct PrimaConfiguration {
     from_addr: String,
     namespace: String,
     environment: Environment,
+    tags: Vec<String>,
 }
 
 impl PrimaConfiguration {
     pub fn new(to_addr: &str, from_addr: &str, namespace: &str, environment: Environment) -> Self {
+        let env_str = environment.to_string();
         Self {
             to_addr: to_addr.to_string(),
             from_addr: from_addr.to_string(),
             namespace: namespace.to_string(),
             environment,
+            tags: vec![format!("env:{}", env_str)],
         }
     }
+
+    pub fn with_tag<T: std::fmt::Display>(mut self, key: &str, value: &T) -> Self {
```

Comment:
Merged before I saw but I stand by my argument - we don't want people passing the literals here, because we only need the ref, and `&T` -> `T` makes this non obvious (and library api's should be obvious) - sure we could document it but what's better than docs? IDE hints and type signatures (imo)